kotlintest-assertions/src/commonMain/kotlin/io/kotlintest/properties/gens.kt

```diff
@@ -585,3 +585,26 @@ fun Gen.Companion.factors(k: Int): Gen<Int> = object : Gen<Int> {
          .filter { k % it == 0 }
    }
 }
+
+/**
+ * Returns a [[Gen]] which returns the sample values in same order as they are passed in, once all sample values are used
+ * it repeats elements from starting.
+ */
+fun <T:Any> Gen.Companion.samples(vararg sampleValues: T) = object : Gen<T> {
```

Comment:
This should probably be `<T: Any?>`, right? We might want nullable values in the sample